### PR TITLE
Add KeyBee65 Firmware

### PR DESCRIPTION
**Pull request by ToastyStoemp**
## Description

Add the firmware for the KeyBee65 PCB used in the Pollen65

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keybee/keybee65/config.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+// The pin connected to the data pin of the LEDs
+#define RGB_DI_PIN B0
+// The number of LEDs connected
+#define DRIVER_LED_TOTAL 80
+
+#define RGB_MATRIX_KEYPRESSES
+#define USB_LED_INDICATOR_ENABLE
```
Comment (drashna) [inline keyboards/keybee/keybee65/config.h:61]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keybee/keybee65/readme.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+---
+
+* Keyboard Maintainer: [ToastyStoemp](https://github.com/ToastyStoemp)
+* Hardware Supported: Pollen65
+* Hardware Availability: [keybeeshop.com](https://www.keybeeshop.com/collections/keyboards/products/pollen65-mechanical-keyboard-kit)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make keybee/keybee65:default
+    make keybee/keybee65:via
```
Comment (fauxpark) [inline keyboards/keybee/keybee65/readme.md:20]:
```suggestion
```

File: keyboards/keybee/keybee65/info.json [hunk 3]
```diff
@@ -0,0 +1,363 @@
+{
+    "keyboard_name": "KeyBee65",
+    "url": "",
+    "maintainer": "ToastyStoemp",
+    "width": 16,
+    "height": 5,
```
Comment (fauxpark) [inline keyboards/keybee/keybee65/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/keybee/keybee65/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        default:
+            return true; //Process all other keycodes normally
+    }
+}
+
+void rgb_matrix_indicators_kb(void) {
+    led_t host_leds = host_keyboard_led_state();
+    if (host_leds.caps_lock) {
+        rgb_matrix_set_color(30, 254, 189, 41); // set caps lock led color first nunber is index, R G B
+    }
```
Comment (drashna) [inline keyboards/keybee/keybee65/keymaps/via/keymap.c:102]: To clean up when the enabled flags don't contain the modifier flag.
```suggestion
    } else if (!(rgb_matrix_get_flags() & LED_FLAG_MODIFIER)) {
        rgb_matrix_set_color(30, 0, 0, 0);
    }
```

File: keyboards/keybee/keybee65/readme.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+* RGB underglow
+* Per-Key RGB
+
+---
+
+* Keyboard Maintainer: [ToastyStoemp](https://github.com/ToastyStoemp)
+* Hardware Supported: Pollen65
+* Hardware Availability: [keybeeshop.com](https://www.keybeeshop.com/collections/keyboards/products/pollen65-mechanical-keyboard-kit)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/keybee/keybee65/readme.md:17]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keybee/keybee65/rules.mk [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/keybee/keybee65/rules.mk:21]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keybee/keybee65/readme.md (hunk 5 above)
Comment (ToastyStoemp) [reply keyboards/keybee/keybee65/readme.md:17]: Done

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**